### DISCUSSION TITLE
Updated example reports error

### DISCUSSION THREAD

**Discussion by terrypin999** [Makuna/DFMiniMp3#101] [Q&A]
When I verified your recently updated example * on my UNO I got the error
'Serial1' was not declared in this scope. Presumably a very basic error on my part, as I have little C/C++ know how. Can you advise please?

'''
* It's the 128 line sketch, starting with

// this example will play a track and then 
// every five seconds play another track

and ending with

void loop() 
{
  waitMilliseconds(100);
}
'''

**Comment by Makuna** [comment]
(@terrypin999 moved the issue to Discussions where it should have been created)

I have no Idea what sketch you are looking at, but the example sketch doesn't have that text on line 128.  The error happens on line 24, that area of code looks like the following.

```
// define a handy type using serial and our notify class
//
typedef DFMiniMp3<HardwareSerial, Mp3Notify> DfMp3; 

// instance a DfMp3 object, 
//
DfMp3 dfmp3(Serial1); 
```

The UNO doesn't have a second Serial object, thus no Serial1.  This is a common issue with Arduino, not all devices have the common hardware objects.  If you look at lines that follow the one with the error (26-30) then you will find your solution.
```
// Some arduino boards only have one hardware serial port, so a software serial port is needed instead.
// comment out the above definitions and use these
//SoftwareSerial secondarySerial(10, 11); // RX, TX
//typedef DFMiniMp3<SoftwareSerial, Mp3Notify> DfMp3;
// DfMp3 dfmp3(secondarySerial);
```

**Comment by terrypin999** [comment]
Thanks, that explains it. Shame it won't work as it stands on the most popular Arduino board.

This is where I downloaded your sketch, 'PlayMP3.ino':
https://github.com/Makuna/DFMiniMp3/blob/master/examples/PlayMp3/PlayMp3.ino

It has 128 lines and I showed the first and last few in my post.
